src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -669,17 +676,26 @@ EnsureRelationHasCompatibleSequenceTypes(Oid relationId)
  * dependentSequenceList, and then alters the sequence type if not the same with the column type.
  */
 static void
-EnsureDistributedSequencesHaveOneType(Oid relationId, List *dependentSequenceList,
-									  List *attnumList)
+EnsureDistributedSequencesHaveOneType(Oid relationId, List *seqInfoList)
 {
-	AttrNumber attnum = InvalidAttrNumber;
-	Oid sequenceOid = InvalidOid;
-	forboth_int_oid(attnum, attnumList, sequenceOid, dependentSequenceList)
+	SequenceInfo *seqInfo = NULL;
+	foreach_ptr(seqInfo, seqInfoList)
 	{
+		if (!seqInfo->isNextValDefault)
+		{
+			/*
+			 * If a sequence is not on the nextval, we don't need any check.
+			 * This is a dependent sequence via ALTER SEQUENCE .. OWNED BY col
```

Comment:
can you have sequences that appear in nextval, but are not dependent?

Reply:
I think not, and I also started to think that the `default value` dependency is probably required to prevent such cases via `ALTER SEQUENCE .. OWNED BY NONE`. As soon as there is a next val, pg add a `default value` dependency, and that dependent for the column
```SQL
┌────────────────────────────────────────┐
│     pg_identify_object_as_address      │
├────────────────────────────────────────┤
│ ("default value","{public,test,b}",{}) │
└────────────────────────────────────────┘
(1 row)
```

Reply:
What about cases where the nextval is part of a bigger expression (e.g. `fn(nextval('seq')))`?

Reply:
The situation is the same, the seq depends on a `default value` and there seems no API (other than DROP CASCADE seq) that breaks this dependency. And, drop cascades to `default value` in the end
```SQL
CREATE SEQUENCE sc1;
CREATE TABLE test(a int, b bigint default double(nextval ('sc1')));
ALTER SEQUENCE sc1 OWNED BY NONE;
 drop sequence sc1;
ERROR:  cannot drop sequence sc1 because other objects depend on it

 drop sequence sc1 cascade;
NOTICE:  drop cascades to default value for column b of table test
```